conn.go

```diff
@@ -479,7 +479,11 @@ func (c *Conn) readLoop() {
 		}
 
 		frameType, data, err := ReadUnpackedResponse(c)
+
 		if err != nil {
+			if err == io.EOF && atomic.LoadInt32(&c.closeFlag) == 1 {
+				goto exit
+			}
 			if !strings.Contains(err.Error(), "use of closed network connection") {
 				c.log(LogLevelError, "IO error - %s", err)
 				c.delegate.OnIOError(c, err)
```

Comment:
@mreiferson @jehiah Should it check `c.closeFlag` if the error does contain "use of closed network connection"? Otherwise, it might skip a call to `c.delegate.OnIOError` which delegates to `(p *Producer) close()`. It's not immediately obviously what conditions inside the `net` package can lead to `net.errClosing`.